File: src/main/java/com/ottamotta/yunar/Pony.java
package com.ottamotta.yunar;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.time.Instant;
import java.time.temporal.ChronoUnit;

@Entity
@Table(name = "ponies")
public class Pony {

    @Id
    String name;

    @Column
    boolean available;

    @Column(name = "returned_timestamp")
    long returnedTimestamp;

    @Column(name = "recharge_time_mins")
    int rechargeTimeMins;


    public String getName() {
        return name;
    }

    public boolean isAvailable() {
        return available;
    }

    public long getReturnedTimestamp() {
        return returnedTimestamp;
    }

    public int getRechargeTimeMins() {
        return rechargeTimeMins;
    }

    public void setAvailable(boolean available) {
        this.available = available;
    }

    public void setReturnedTimestamp(long returnedTimestamp) {
        this.returnedTimestamp = returnedTimestamp;
    }

    public boolean isRecharged() {
        return returnedTimestamp < 0 ||
                Instant.ofEpochMilli(returnedTimestamp)
                        .plus(rechargeTimeMins, ChronoUnit.MINUTES)
                        .isAfter(Instant.ofEpochMilli(System.currentTimeMillis()));
    }
}

File: src/main/kotlin/com/ottamotta/yunar/Pony.kt
package com.ottamotta.yunar

import com.fasterxml.jackson.annotation.JsonIgnore
import java.time.Instant
import java.time.temporal.ChronoUnit
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table
import javax.persistence.Transient

@Entity
@Table(name = "ponies")
data class Pony (

    @Id
    val name: String,

    @Column(name = "available")
    val isAvailable: Boolean = false,

    @Column(name = "returned_timestamp")
    val returnedTimestamp: Long = 0,

    @Column(name = "recharge_time_mins")
    val rechargeTimeMins : Int) {

    @JsonIgnore
    fun isRecharged() : Boolean = returnedTimestamp < 0 || Instant.ofEpochMilli(returnedTimestamp)
                .plus(rechargeTimeMins.toLong(), ChronoUnit.MINUTES)
                .isBefore(Instant.ofEpochMilli(System.currentTimeMillis()))
}